domain/tests/src/Unit/DomainValidatorTest.php

```diff
@@ -0,0 +1,126 @@
+use Drupal\domain\DomainValidator;
+
+/**
+ * Tests domain record validation.
+ *
+ * @group domain
+ */
+class DomainValidatorTest extends UnitTestCase {
+
+  /** @var \Drupal\domain\DomainValidator */
+  private $object;
+
+  /** @var \GuzzleHttp\Client */
+  public $client;
+
+  /**
+   * Set up a service container and SUT (Subject Under Test).
+   *
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    /** @var \Drupal\Core\Config\ImmutableConfig $config */
+    $config = $mock = $this->getMockBuilder('Drupal\Core\Config\ImmutableConfig')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('get')
+      ->willReturnMap([
+        ['allow_non_ascii', FALSE],
+        ['www-prefix', FALSE]
+      ]);
+
+    /** @var \Drupal\Core\Config\ConfigFactory $factory */
+    $factory = $mock = $this->getMockBuilder('Drupal\Core\Config\ConfigFactory')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::once())->method('get')->willReturn($config);
+
+    /** @var \Drupal\Core\Entity\EntityStorageBase $storage */
+    $storage = $mock = $this->getMockBuilder('Drupal\Core\Entity\EntityStorageBase')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('loadByProperties')->willReturn(array());
+
+    /** @var \Drupal\Core\Entity\EntityTypeManager $manager */
+    $manager = $mock = $this->getMockBuilder('Drupal\Core\Entity\EntityTypeManager')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('getStorage')
+      ->willReturn($storage);
+
+    /** @var \Drupal\Core\Extension\ModuleHandler $handler */
+    $handler = $mock = $this->getMockBuilder('Drupal\Core\Extension\ModuleHandler')
+      ->disableOriginalConstructor()->getMock();
+
+    $response = $this->getMockBuilder('GuzzleHttp\Psr7\Response')
+      ->disableOriginalConstructor()->getMock();
+    $response->expects(static::any())->method('getStatusCode')
+      ->willReturn(200);
+
+    $client = $this->getMockBuilder('GuzzleHttp\Client')->setMethods(['request'])
+      ->disableOriginalConstructor()->getMock();
+    $client->expects(static::any())->method('request')
+      ->will(static::returnValue($response));
+
+    $this->object = new DomainValidator($handler, $factory, $client, $manager);
+  }
+
+  /**
+   * @param string $host
+   * @param int $result
+   *
+   * @dataProvider provider
+   */
+  public function testValidateStaticInspection($host, $result) {
+    /** @var \Drupal\domain\Entity\Domain $domain */
+    $domain = $mock = $this->getMockBuilder('Drupal\domain\Entity\Domain')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('getHostname')
+      ->willReturn($host);
+    $errors = $this->object->validate($domain);
+    static::assertNotEquals((bool) $result, (bool) count($errors));
+  }
+
+  /**
+   * @return array
+   */
+  public function provider() {
```

Comment:
The convention is providerAspectUnderTest provides data for testAspectUnderTest. In starting out, I had tested the static methods directly before making making it private. When I ran the same test data through both the public method and the (now private) internal method, I left it more generic. I will update to match the test name as is usually the case.

expectedResult() seems less clear because the data provider contains both the result and the data that produce the result. I've never seen it used as a name for a data provider.

The name is only a convention, The @dataProvider annotation invokes the method, so any method name works as long as it is the name in the annotation.